src/sentry/web/frontend/accounts.py

```diff
@@ -106,14 +106,16 @@ def recover_confirm(request, user_id, hash, mode="recover"):
             password_hash.delete()
             raise LostPasswordHash.DoesNotExist
         user = password_hash.user
-
     except LostPasswordHash.DoesNotExist:
         return render_to_response(get_template(mode, "failure"), {}, request)
 
+    form = RelocationForm if mode == "relocate" else ChangePasswordRecoverForm
```

Comment:
My 2c: we should decide this based on the user's `is_unclaimed` status, not which email link they used. Even if some wires get crossed in the emails (or, more likely, the user tries to hit the URL directly), we should still show them the correct form. Wdyt?

Reply:
Ah that's a good point. Currently, the setup allows wires to get crossed in the emails (I just attempted to hit the URL routes directly on sentry.io). I did some digging, and it's impossible to know whether or not the mode should be `set_password` or `recover` if they try to hit the relocate url. So, I think this is fine for now

Reply:
Sgtm, though I think we should still leave a TODO (maybe pointed at `getsentry/team-ospo#190`?) so that we remember to circle back to that later.

Reply:
sure